src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,544 @@
+        self.dwellTime = model.FloatContinuous(float(self.clockPeriod.value), (float(self.clockPeriod.value), 100000.0),
+                                               unit='s')
+        self.pixelSize = model.TupleContinuous((4, 4), range=((1, 1), (100000, 100000)), unit='px,px',
+                                               setter=self._setpixelSize)
+        self.rotation = model.FloatContinuous(0, range=(0, 2 * numpy.pi), unit='rad, cw=+')
+        self.scanFlyback = model.FloatVA(0, unit='s')
+        self.scanOffset = model.TupleContinuous((0.0, 0.0), range=((-10.0, -10.0), (10.0, 10.0)), unit='V,V')
+        self.scanGain = model.TupleContinuous((0.0, 0.0), range=((-10.0, -10.0), (10.0, 10.0)), unit='V,V')
+        self.scanDelay = model.TupleContinuous((0, 0), range=((1, 1), (100000, 100000)), unit='s,s',
+                                               setter=self._setScanDelay)
+
+        self._metadata[model.MD_PIXEL_SIZE] = self.pixelSize.value
+        self._metadata[model.MD_DWELL_TIME] = self.dwellTime.value
+
+    def _setResolution(self, input_resolution):
+        if input_resolution[0] == input_resolution[1]:
+            return input_resolution
+        else:
+            logging.warning("Non-square resolution entered, only square resolution sizes are supported")
+            return self.resolution.value
+
+    def _setpixelSize(self, input_resolution):
+        if input_resolution[0] == input_resolution[1]:
+            return input_resolution
+        else:
+            logging.warning("Non-square pixel size entered, only square pixel sizes are supported")
+            return self.pixelSize.value
+
+    def _setScanDelay(self, input_delay):
+        if not(hasattr(self.parent, "_MPPC")) or self.parent._MPPC.acqDelay.value - input_delay[0] >= 0:
+            return input_delay
+        else:
+            logging.error("Wrong input x_scan_to_acq_delay got a negative value.\n"
+                          "Change values so that 'self.parent._MPPC. acqDelay.value - self.scanDelay.value[0]' has a positive result")
+            return self.scanDelay.value
+
+class MirrorDescanner(model.Emitter):
+
+    def __init__(self, name, role, parent, **kwargs):
+        super(MirrorDescanner, self).__init__(name, role, **kwargs)
+        self.parent = parent
+        # model.Emitter(name, role)
+        self.rotation = model.FloatContinuous(0, range=(0, 2 * numpy.pi), unit='rad, cw=+')
+        self.scanOffset = model.TupleContinuous((0.0, 0.0), range=((-10.0, -10.0), (10.0, 10.0)), unit='V,V')
+        self.scanGain = model.TupleContinuous((0.0, 0.0), range=((-10.0, -10.0), (10.0, 10.0)), unit='V,V')
+
+
+class MPPC(model.Detector):
+    def __init__(self, name, role, parent, **kwargs):
+        super(MPPC, self).__init__(name, role, **kwargs)
+        self.parent = parent
+
+        if not (hasattr(self.parent, "_MirrorDescanner")) or not (hasattr(self.parent, "_EBeamScanner")):
```

Comment:
If somewhere this class is inhertited or used for something else this check gives a clear error. While if this check if not performed weird error will come because these sister objects are needed. In the parent class such a check is not needed I think because there all the childeren are created during construction (__init__) of the object.

Reply:
Then I would suggest to add a comment 
`Descanner needs to be there before instantiating the detector.`